202312-quarto-reports/2023-12-quarto-reports.qmd

```diff
@@ -0,0 +1,278 @@
+
+## How do I turn an .Rmd into a Quarto document?
+
+Example report document available here: <https://github.com/cct-datascience/workshop-wednesdays/blob/2023-quarto-reports/20231206-quarto-reports/mouse_report_RMD.Rmd>
+
+
+## Technical changes from .Rmd to Quarto
+
+::: incremental
+**Most .Rmd documents will render as-is!**
+
+-   Document-level options moved to header
+
+-   `output` changed to `format`
+
+-   Chunk-level options moved from ```` ```{r ....} ```` to `#|` within the chunk
+
+-   `knitr::convert_chunk_header(<file_path>)` helps with conversion
+
+-   <https://quarto.org/docs/faq/rmarkdown.html>
+:::
+
+
+## .Rmd to .Qmd
+
+::: incremental
+
+1. Run `knitr::convert_chunk_header`
+1. Add additional Quarto features 
+
+:::
+
+## Sharing Quarto reports
+
+| Hosted online with a shareable URL | Static documents to share person-to-person |
+|:-----------------------------------|:-----------------------------------|
+| Quarto Pub                         | Rendered HTML, pdf, docx outputs           |
+| Posit Connect                      | Source .qmd files                          |
+| GitHub                             |                                            |
+
+## QuartoPub
+
+::: incremental
+-   <https://quartopub.com/>
+
+-   Create a free account
+
+-   Published reports are **public**
+
+-   Example: <https://cct-datascience.quarto.pub/quarto-workshop-example/>
+:::
+
+## Posit Connect
+
+::: incremental
+-   <https://viz.datascience.arizona.edu>
+
+-   For **university research**; accounts available but restricted
+
+-   Published reports can be public or private
+
+-   Example: <https://viz.datascience.arizona.edu/quarto-workshop-example/>
+:::
+
+## GitHub
+
+::: incremental
+-   Render using the `gfm` output format
+
+-   Resulting .md files are visible online on GitHub
+
+-   Published reports may be public or private (depending on repo scopes)
+
+-   Example: <https://github.com/diazrenata/squareone-quarto/blob/main/s3_biomass_as_quarto.md>
+:::
+
+## Sharing rendered documents
+
+::: incremental
+-   Render to `html`, `docx`, or `pdf`
+
+-   Resulting documents are portable
+:::
+
+## Sharing source documents
+
+::: incremental
+-   Share, edit, and re-render a single .qmd
+
+-   Especially for collaboration on GitHub
+
+-   Example: <https://github.com/diazrenata/squareone-quarto/blob/main/s3_biomass_as_quarto.qmd>
+:::
+
+# Break!
+
+## Practice time: Create a Quarto report from your script
+
+## Practice time: Share your report
+
+Pick one (or more!) of the following options:
+
+::: incremental
+-   Create a QuartoPub account and publish your report
+-   Render your report as a GitHub document and push it to your GitHub
+-   Render your report as an HTML file and email it to yourself
+:::
+
+# Quarto resources
+
+## Quarto docs
+
+::: incremental
+-   <https://quarto.org>
+
+-   <https://quarto.org/docs/guide/>
+
+-   Cheatsheet: <https://rstudio.github.io/cheatsheets/quarto.pdf>
+:::
+
+## CCT-Data Science
```

Comment:
You might want to mention our group and show our website briefly at the beginning too, for folks who drop off.